arangod/Scheduler/Scheduler.cpp

```diff
@@ -283,15 +340,21 @@ bool Scheduler::queue(RequestPriority prio,
 
 void Scheduler::drain() {
   while (canPostDirectly()) {
-    bool found = popFifo(1);
+    bool found = popFifo(FIFO1);
 
     if (!found) {
-      found = popFifo(2);
+      found = popFifo(FIFO8);
 
       if (!found) {
-        break;
+        found = popFifo(FIFO2);
+      } else if (canPostDirectly()) {
+        popFifo(FIFO2);
```

Comment:
It seems we are only getting here when we have successfully popped from `FIFO8` already.
Is it actually required that we additionally pop from `FIFO2` then?